### ISSUE TITLE
boxed_local false positive for ?Sized

### ISSUE THREAD

**Issue by CAD97** [rust-lang/rust-clippy#4740]
<kbd>L-bug :beetle:</kbd> <kbd>L-suggestion-causes-error</kbd>

[Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=0e544b2ee90b4ea682a0913e7629530e)

```rust
fn lol(x: Box<[u8]>) {
    unimplemented!();
}
```

```
warning: local variable doesn't need to be boxed here
 --> src/lib.rs:1:8
  |
1 | fn lol(x: Box<[u8]>) {
  |        ^
  |
  = note: `#[warn(clippy::boxed_local)]` on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#boxed_local
```

This is kind of an interesting case; the suggestion isn't _wrong_ per se, but it's definitely not optimal. The lint explainer as currently worded strongly suggests "an unboxed `T` would work fine", whereas with `?Sized` values this obviously cannot be done. The owned value has to be boxed or otherwise behind an indirection.

**Comment by flip1995** [comment]
The lint triggering here is correct, we should suggest to turn this into `&[u8]` though. See https://github.com/rust-lang/rust-clippy/issues/4351#issuecomment-524600717
